e2e_test/udf/python.slt

```diff
@@ -123,6 +123,34 @@ select (extract_tcp_info(E'\\x45000034a8a8400040065b8ac0a8000ec0a80001035d20b6d9
 ----
 192.168.0.14 192.168.0.1 861 8374
 
+# error handling
+
+statement error
+select hex_to_dec('1z');
+
+statement ok
+create table t (dummy date, v varchar);
+
+statement ok
+create materialized view mv as select dummy, hex_to_dec(v) from t;
+
+statement ok
+insert into t values ('2023-01-01', '1z');
```

Comment:
With the dummy column, it triggers panic:
```
thread 'risingwave-streaming-actor' panicked at 'Failed to append datum, array builder type: Utf8, scalar type: Date', src/common/src/array/mod.rs:562:1
```

Without the dummy column, it triggers panic:
```
thread 'risingwave-streaming-actor' panicked at 'itertools: .zip_eq() reached end of one iterator before the other', /Users/xiangjin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/itertools-0.10.5/src/zip_eq_impl.rs:48:13
```